[jack1e0] iP

jack1e0 opened the pull request:
### BobPro
> “Bob.” – Bob ([source](https://dictionary.cambridge.org/dictionary/english/bob))

BobPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~  **_SUPER_** FAST to use

All you need to do is:
1. Download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. Double click it
3. Add your tasks
4. Let it manage them for you!😆 

And it's **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
##
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jack1e0 committed:
incr: level 0

jack1e0 committed:
wip

jack1e0 committed:
incr: level 1

jack1e0 committed:
Merge remote-tracking branch 'origin/master'

jack1e0 committed:
incr: level 1

jack1e0 committed:
incr: level 2

jack1e0 committed:
incr: level 3

jack1e0 committed:
incr: level 4

jack1e0 committed:
new incr: level 4

jack1e0 committed:
incr: testing

jack1e0 committed:
incr: error handling

jack1e0 committed:
incr: level 6

jack1e0 committed:
incr: enums

jack1e0 committed:
debug

jack1e0 committed:
renaming

jack1e0 committed:
renaming

jack1e0 committed:
incr: level 7

jack1e0 committed:
Merge branch 'branch-Level-7'

jack1e0 committed:
incr: level 8

jack1e0 committed:
Merge branch 'branch-Level-8'

jack1e0 committed:
more oop

jack1e0 committed:
more oop (commands)

jack1e0 committed:
Merge branch 'A-MoreOOP'

jack1e0 committed:
package separation

jack1e0 committed:
renaming

jack1e0 committed:
Merge branch 'A-Packages'

jack1e0 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jack1e0 committed:
gradle support

jack1e0 committed:
debugging

jack1e0 committed:
jar

jack1e0 committed:
javadocs

jack1e0 committed:
coding standards

jack1e0 committed:
incr: level 9

jack1e0 committed:
Merge branch 'branch-A-JavaDoc'

jack1e0 committed:
Merge branch 'branch-A-CodingStandard'

jack1e0 committed:
Merge branch 'branch-Level-9'

jack1e0 committed:
junit

jack1e0 committed:
checkstyle

jack1e0 committed:
Merge branch 'branch-A-Checkstyle'

jack1e0 committed:
javafx basics

jack1e0 committed:
javafx tut 2

jack1e0 committed:
scrolling as intended

jack1e0 committed:
format dialogue box

jack1e0 committed:
better ui

jack1e0 committed:
using fxml

jack1e0 committed:
making MainWindow fx:root

jack1e0 committed:
bug: stage

jack1e0 committed:
no bug, no stage

jack1e0 committed:
integrate gui

jack1e0 committed:
Merge branch 'branch-Level-10'

jack1e0 committed:
cleaning

NgChunMan commented on src/main/java/bob/task/Todo.java:
```diff
@@ -0,0 +1,48 @@
+     */
+    @Override
+    public String toString() {
+        String done = this.done ? "[X]" : "[ ]";
```
Perhaps a variable name such as status which is a noun would be more direct?

NgChunMan commented on src/main/java/bob/command/MarkCommand.java:
```diff
@@ -0,0 +1,55 @@
+                doneOrNot = false;
+            }
+
+            String display = ui.stringFormat(tasks.markDoneOrNot(index, doneOrNot));
```
Perhaps the variable name textDisplayed would be easier to understand?

NgChunMan commented on src/main/java/bob/exception/BobException.java:
```diff
@@ -0,0 +1,9 @@
+ * Contains exceptions specific to Bob
+ */
+public class BobException extends Exception {
+    public String message = "Error.";
```
Perhaps errorMessage as the variable name would be better?

NgChunMan commented on src/main/java/bob/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+ */
+public class TaskList {
+
+    public ArrayList<Task> lst;
```
Perhaps tasksList as the variable name would be better?

tiongjjyi replied:
Maybe you may want to consider a more intuitive name on the list?

NgChunMan reviewed (commented):
LGTM. The code exhibits good code quality. Just a few nits to fix.

s-peiran commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,66 @@
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
```
Good on separating the different import classes

s-peiran commented on src/main/java/bob/DialogBox.java:
```diff
@@ -0,0 +1,69 @@
+package bob;
+
+import java.io.IOException;
+
```
good space to separate the imports

s-peiran commented on src/main/java/bob/MainWindow.java:
```diff
@@ -0,0 +1,56 @@
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
```
Good! private modifier for abstraction

s-peiran commented on src/main/java/bob/MainWindow.java:
```diff
@@ -0,0 +1,56 @@
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        // dialogContainer.getChildren().add(DialogBox.getBobDialog(bob.greet(), bobImage));
```
Delete unwanted code

s-peiran commented on src/main/java/bob/command/ExitCommand.java:
```diff
@@ -0,0 +1,21 @@
+public class ExitCommand extends Command {
+
+    public ExitCommand() {
+        super.isExit = true;
```
maybe hasExit is a better variable name

s-peiran reviewed (commented):
Well formatted code. Consider some suggestions below

tiongjjyi commented on src/main/java/bob/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+     * @throws MissingDatesException when start and end of Event is missing.
+     */
+    public Task generateTask(String description)
+            throws WrongInputException, MissingTaskException,
```
I like how you split the line into two to improve readability.

tiongjjyi commented on src/main/java/bob/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+            MissingDatesException, DateTimeParseException {
+
+        // Split by the first " " into type, and task details
+        String[] task = description.split(" ", 2);
```
Good job on following the coding standard that the array specifier is attached to the type.

tiongjjyi commented on src/main/java/bob/command/Command.java:
```diff
@@ -0,0 +1,21 @@
+
+    protected String input;
+
+    protected boolean isExit;
```
This boolean variable does sound like boolean!

tiongjjyi commented on src/main/java/bob/exception/MissingDatesException.java:
```diff
@@ -0,0 +1,9 @@
+ * for Events and Deadlines
+ */
+public class MissingDatesException extends BobException {
+    public String message = "Please provide the appropriate date!";
```
Perhaps you could use a more specific naming on this variable?

tiongjjyi reviewed (commented):
Amazing work! Majority of the codes followed the coding standard.

jack1e0 committed:
Create new jar file

jack1e0 committed:
Add assertions

Assertions can be used to identify bugs in the code.

Let's incorporate assertions in different points in the code.

jack1e0 committed:
Improve code quality

Refactors different parts of code to improve readability and abide by coding standards.

jack1e0 committed:
Add streams

Converted for-loops into streams

jack1e0 committed:
Merge pull request #1 from jack1e0/branch-A-Assertions

Add assertions

jack1e0 committed:
Merge branch 'master' into branch-A-Streams

jack1e0 committed:
Merge branch 'master' into branch-A-CodeQuality

jack1e0 committed:
Merge pull request #2 from jack1e0/branch-A-CodeQuality

Improve code quality

jack1e0 committed:
Merge branch 'master' into branch-A-Streams

jack1e0 committed:
Merge pull request #3 from jack1e0/branch-A-Streams

Add streams

jack1e0 committed:
Set up CI

jack1e0 committed:
Add priority field for tasks

jack1e0 committed:
Display different priorities

jack1e0 committed:
Merge branch 'branch-C-Priority'

jack1e0 committed:
Better GUI

jack1e0 committed:
Allow resizing

jack1e0 committed:
Personality

jack1e0 committed:
Handle white spaces

jack1e0 committed:
Handle invalid dates, and repeated tasks

jack1e0 committed:
Debug

jack1e0 committed:
Product screenshot

jack1e0 committed:
Add user guide

jack1e0 committed:
Update user guide

jack1e0 committed:
Update README.md

jack1e0 committed:
Update Ui.png

jack1e0 committed:
Merge remote-tracking branch 'origin/master'

jack1e0 committed:
bug

jack1e0 committed:
debug
